test/e2e/subroutes_test.go

```diff
@@ -109,66 +110,285 @@ func TestSubrouteVisibilityChange(t *testing.T) {
 	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
 	defer test.TearDown(clients, names)
 
-	tag := "my-tag"
+	subrouteTag1 := "my-tag"
+	subrouteTag2 := "my-tag2"
 
-	withInternalVisibility := WithServiceLabel(
-		routeconfig.VisibilityLabelKey, routeconfig.VisibilityClusterLocal)
+	withTrafficSpec := WithInlineRouteSpec(v1alpha1.RouteSpec{
+		Traffic: []v1alpha1.TrafficTarget{{
+			TrafficTarget: v1.TrafficTarget{
+				Tag:     subrouteTag1,
+				Percent: ptr.Int64(100),
+			},
+		}},
+	})
+	resources, err := v1a1test.CreateRunLatestServiceReady(t, clients, &names, withTrafficSpec)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
+	if isClusterLocal, err := trafficWithTagIsClusterLocal(resources.Route.Status.Traffic, subrouteTag1); err != nil {
+		t.Fatalf(err.Error())
+	} else if isClusterLocal {
+		t.Fatalf("Expected subroutes with tag %s to be not cluster local", subrouteTag1)
+	}
+
+	if isRouteClusterLocal(resources.Route.Status) {
+		t.Fatalf("Expected route to be not cluster local")
+	}
+
+	serviceName := fmt.Sprintf("%s-%s", subrouteTag1, resources.Route.Name)
+	svc, err := clients.KubeClient.Kube.CoreV1().Services(test.ServingNamespace).Get(serviceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get k8s service to modify: %s", err.Error())
+	}
+
+	svcCopy := svc.DeepCopy()
+	labels.SetVisibility(&svcCopy.ObjectMeta, true)
+
+	svcpatchBytes, err := test.CreateBytePatch(svc, svcCopy)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	if _, err = clients.KubeClient.Kube.CoreV1().Services(test.ServingNamespace).Patch(serviceName, types.JSONPatchType, svcpatchBytes); err != nil {
+		t.Fatalf("Failed to patch service: %s", err.Error())
+	}
+
+	//	create sub-route 2 in kservice
+	ksvcCopy := resources.Service.DeepCopy()
+	ksvcCopyRouteTraffic := append(ksvcCopy.Spec.Traffic,
+		v1alpha1.TrafficTarget{
+			TrafficTarget: v1.TrafficTarget{
+				Tag:            subrouteTag2,
+				LatestRevision: ptr.Bool(true),
+			},
+		})
+
+	if _, err = v1a1test.UpdateServiceRouteSpec(t, clients, names, v1alpha1.RouteSpec{Traffic: ksvcCopyRouteTraffic}); err != nil {
+		t.Fatalf("Failed to patch service: %s", err.Error())
```

Comment:
Same for all the errors below. 

```suggestion
		t.Fatalf("Failed to patch service: %w", err.)
```